Add 21.09 release announcement (#994)
galaxyproject/galaxy-hub: content/news/2021-11-galaxy-release-21-09/index.md

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/aPgMcTuzEJ4" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

Additionally, did a *collection ever fail* for you? And you wondered why? Now you can find out with the **view details button for collections** ([Pull Request 12261](https://github.com/galaxyproject/galaxy/pull/12261))!

### Importing Data

Selecting datasets from the Remote Files view has gotten easier! Now you can select folders and files, and it will import all of the datasets recursively in those folders. Previously you could only select files within a folder so this is a huge improvement in usability of such a key new feature of Galaxy ([Pull Request 12310](https://github.com/galaxyproject/galaxy/pull/12310)). But with great power comes great responsibility, be careful not to import the entirety of NCBI!

<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/XcZgQbGojh8" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

### Reports

> Report components used to try and arrange themselves in a smart way, but due to some issues with the floating of report components these will now be full width until we can figure out a good way to provide a similar feature allowing you to control the layout of the report ([Pull Request 12085](https://github.com/galaxyproject/galaxy/pull/12085)).

Reviewer: ```suggestion
Report components used to automatically arrange themselves in a smart way, but due to some issues with the floating of components, these will now be full width until we can figure out a good way to provide a similar feature, allowing you to control the layout of the report ([Pull Request 12085](https://github.com/galaxyproject/galaxy/pull/12085)).
```

Reviewer: `allowing you to control the layout of the report` is a bit confusing to me: Does it mean that in the future users will be able to arrange the components or is it happening already?

Author: In the future, I think we can just drop that comment.

![A screenshot of the reports interface is shown, there are a number of 'report components' like table previews embedded and these are full width across the report.](images/21.09-reports.png)

> ### Enhancements for working with remote data and distributed computing resources

Reviewer: ```suggestion
### Enhancements for Working with Remote Data and Distributed Computing Resources
```

Many fixes and enhancements were made to improve how Galaxy can import and
write to remote data using the configured file source plugins. In addition to
the History Export functionality, which can write to remote locations, we have
added an ``Export datasets`` Tool for exporting individual Datasets and Dataset
collections to configured remote locations. The Tool will automatically
maintain the name and structure of Datasets and Dataset Collections. We have
added the possibility to import entire folders of remote data and made the
data selection dialog more intuitive. We also added numerous improvements and
fixes to the extended metadata collection strategy and [Pulsar](https://github.com/galaxyproject/pulsar),
so that many more Tools are able to run in [Pulsar](https://github.com/galaxyproject/pulsar)
and write data back to storage without passing through Galaxy first.

> ### Migration to FastAPI and extended Documentation of API routes

Reviewer: ```suggestion
### Migration to FastAPI and Extended Documentation of API Routes
```

We have modernized and migrated many more API routes to FastAPI and have
extended the documentation and validation of these routes.

### Migration to SQLAlchemy 1.4 and Declarative Mapping

Galaxy is now using [SQLAlchemy](https://www.sqlalchemy.org/) version 1.4,
which is a prerequisite for asynchronously interacting with the Database. We
also modernized the way our Database models are defined to the more commonly
used Declarative Mapping approach, which is more concise and better
documented.

> ### Modernization of Tool Form interface

Reviewer: ```suggestion
### Modernization of Tool Form Interface
```

> The Tool Form interface has been almost completely migrated from

Reviewer: no sure about this capitalization/lower cases, please ignore if it's wrong

[Backbone](https://backbonejs.org) to [Vue](https://vuejs.org/).
This improves the reactivity of parameter validation and enables the migration of
Tool parameters to Vue, which will also allow us to add many more types of
Tool parameters. We further migrated the Tool Form variant for the Workflow

> Editor, the Workflow Run Form and the Show Dataset Parameter page to Vue.

Reviewer: ```suggestion
Editor, the Workflow run form and the Show Dataset Parameter page to Vue.
```

### New User Welcome Page

> We have added a Page that new users will be directed to after creating a new Galaxy account.

Reviewer: ```suggestion
We have added a page that new users will be directed to after creating a new Galaxy account.
```
